Error when using multiple DepthImageOctomapUpdater in sensors.yaml

### Description
In principle, using multiple  `occupancy_map_monitor/DepthImageOctomapUpdater`  in `sensors.yaml` works fine, when ignoring the errors. Debugging is not easy, since multiple nodes publish on the same topic resulting in a flickering image.
However, moveit_grasps does not seem to like the errors and fails to plan the motion. When using only one sensor, moveit_grasps works just fine.

I coded a really hacky fix, which just appends the `filtered_cloud_topic_` to `model_depth` and `filtered_label` topics: https://github.com/rwth-irt/moveit/commit/e9ed86a8330960b3379c485e1f9ceb54903665f7

Cleaner would either be namespacing all the topics `filtered_depth`, `model_depth` and `filtered_label`.
For example, we could use the `filtered_cloud_topic` which would result in the following topic changes:

- `filtered_cloud_topic` -> `filtered_cloud_topic/filtered_depth`
- `model_depth` -> `filtered_cloud_topic/model_depth`
- `filtered_label` -> `filtered_cloud_topic/filtered_label`

As I understood, these topics are intended for debug only, so I guess it would not hurt too much.
Less breaking could be adding an additional parameter to sensors.yaml, for example `debug_namespace` resulting in:

- `filtered_cloud_topic` -> `filtered_cloud_topic`
- `model_depth` -> `debug_namespace/model_depth`
- `filtered_label` -> `debug_namespace/filtered_label`

### Your environment
* ROS Distro: [Melodic]
* OS Version: e.g. Ubuntu 18.04
* Source build
* If source, which branch: melodic-devel, forked for hacky fix: https://github.com/rwth-irt/moveit/tree/melodic-devel

### Steps to reproduce
Add multiple `occupancy_map_monitor/DepthImageOctomapUpdater`  to `sensors.yaml` in my case:

```yaml
# The name of this file shouldn't be changed, or else the Setup Assistant won't detect it
sensors:
  - far_clipping_plane_distance: 20.0
    filtered_cloud_topic: filtered_cloud_rs_sr305
    image_topic: /rs_sr305/aligned_depth_to_color/image_raw
    max_range: 5.0
    max_update_rate: 1.0
    near_clipping_plane_distance: 0.01
    padding_offset: 0.03
    padding_scale: 4.0
    point_cloud_topic: /rs_sr305/points/PointCloud
    point_subsample: 1
    queue_size: 5
    sensor_plugin: occupancy_map_monitor/DepthImageOctomapUpdater
    shadow_threshold: 0.3
    # Used for self-filtering, better too large than too small
  - far_clipping_plane_distance: 5.0
    filtered_cloud_topic: filtered_cloud_zivid
    image_topic: /zivid/depth/image_raw
    max_range: 2.0
    max_update_rate: 1.0
    # Used for self-filtering, better too small than too large
    near_clipping_plane_distance: 0.01
    # Used for self-filtering, inspect filtered_label topic, enable debug in move_group.launch
    padding_offset: 0.03
    padding_scale: 4.0
    point_cloud_topic: /zivid/points
    point_subsample: 1
    queue_size: 5
    sensor_plugin: occupancy_map_monitor/DepthImageOctomapUpdater
    # Inspect filtered_label topic, enable debug in move_group.launch
    shadow_threshold: 0.2
```

### Expected behaviour
No errors in the log, different topics for different sensors, no flicker when viewing the images.

### Actual behaviour
Errors in the log, multiple nodes publish on the same topic, resulting in flickering / quickly switching debug images.

### Backtrace or Console output

```bash
[ERROR] [1629990154.785818071]: Tried to advertise a service that is already advertised in this node [/move_group/model_depth/compressedDepth/set_parameters]
[ERROR] [1629990154.789655034]: Tried to advertise a service that is already advertised in this node [/move_group/model_depth/compressed/set_parameters]
[ERROR] [1629990154.794728093]: Tried to advertise a service that is already advertised in this node [/move_group/model_depth/theora/set_parameters]
[ERROR] [1629990154.815517635]: Tried to advertise a service that is already advertised in this node [/move_group/filtered_label/compressedDepth/set_parameters]
[ERROR] [1629990154.818915169]: Tried to advertise a service that is already advertised in this node [/move_group/filtered_label/compressed/set_parameters]
[ERROR] [1629990154.825915996]: Tried to advertise a service that is already advertised in this node [/move_group/filtered_label/theora/set_parameters]
```

--- a/moveit_ros/perception/depth_image_octomap_updater/include/moveit/depth_image_octomap_updater/depth_image_octomap_updater.h
+++ b/moveit_ros/perception/depth_image_octomap_updater/include/moveit/depth_image_octomap_updater/depth_image_octomap_updater.h
@@ -80,6 +80,7 @@ private:
   ros::Time last_update_time_;
 
   std::string filtered_cloud_topic_;
+  std::string ns_;
   std::string sensor_type_;
   std::string image_topic_;
   std::size_t queue_size_;

--- a/moveit_ros/perception/depth_image_octomap_updater/src/depth_image_octomap_updater.cpp
+++ b/moveit_ros/perception/depth_image_octomap_updater/src/depth_image_octomap_updater.cpp
@@ -105,6 +105,8 @@ bool DepthImageOctomapUpdater::setParams(XmlRpc::XmlRpcValue& params)
     readXmlParam(params, "skip_horizontal_pixels", &skip_horizontal_pixels_);
     if (params.hasMember("filtered_cloud_topic"))
       filtered_cloud_topic_ = static_cast<const std::string&>(params["filtered_cloud_topic"]);
+    if (params.hasMember("ns"))
+      ns_ = static_cast<const std::string&>(params["ns"]);
   }
   catch (XmlRpc::XmlRpcException& ex)
   {
@@ -135,14 +137,18 @@ bool DepthImageOctomapUpdater::initialize()
 void DepthImageOctomapUpdater::start()
 {
   image_transport::TransportHints hints("raw", ros::TransportHints(), nh_);
-  pub_model_depth_image_ = model_depth_transport_.advertiseCamera("model_depth", 1);
 
+  std::string prefix = "";
+  if (!ns_.empty())
+    prefix = ns_ + "/";
+
+  pub_model_depth_image_ = model_depth_transport_.advertiseCamera(prefix + "model_depth", 1);
   if (!filtered_cloud_topic_.empty())
-    pub_filtered_depth_image_ = filtered_depth_transport_.advertiseCamera(filtered_cloud_topic_, 1);
+    pub_filtered_depth_image_ = filtered_depth_transport_.advertiseCamera(prefix + filtered_cloud_topic_, 1);
   else
-    pub_filtered_depth_image_ = filtered_depth_transport_.advertiseCamera("filtered_depth", 1);
+    pub_filtered_depth_image_ = filtered_depth_transport_.advertiseCamera(prefix + "filtered_depth", 1);
 
-  pub_filtered_label_image_ = filtered_label_transport_.advertiseCamera("filtered_label", 1);
+  pub_filtered_label_image_ = filtered_label_transport_.advertiseCamera(prefix + "filtered_label", 1);
 
   sub_depth_image_ = input_depth_transport_.subscribeCamera(image_topic_, queue_size_,
                                                             &DepthImageOctomapUpdater::depthImageCallback, this, hints);

--- a/moveit_ros/perception/pointcloud_octomap_updater/include/moveit/pointcloud_octomap_updater/pointcloud_octomap_updater.h
+++ b/moveit_ros/perception/pointcloud_octomap_updater/include/moveit/pointcloud_octomap_updater/pointcloud_octomap_updater.h
@@ -87,6 +87,7 @@ private:
   unsigned int point_subsample_;
   double max_update_rate_;
   std::string filtered_cloud_topic_;
+  std::string ns_;
   ros::Publisher filtered_cloud_publisher_;
 
   message_filters::Subscriber<sensor_msgs::PointCloud2>* point_cloud_subscriber_;

--- a/moveit_ros/perception/pointcloud_octomap_updater/src/pointcloud_octomap_updater.cpp
+++ b/moveit_ros/perception/pointcloud_octomap_updater/src/pointcloud_octomap_updater.cpp
@@ -82,6 +82,8 @@ bool PointCloudOctomapUpdater::setParams(XmlRpc::XmlRpcValue& params)
       readXmlParam(params, "max_update_rate", &max_update_rate_);
     if (params.hasMember("filtered_cloud_topic"))
       filtered_cloud_topic_ = static_cast<const std::string&>(params["filtered_cloud_topic"]);
+    if (params.hasMember("ns"))
+      ns_ = static_cast<const std::string&>(params["ns"]);
   }
   catch (XmlRpc::XmlRpcException& ex)
   {
@@ -98,8 +100,13 @@ bool PointCloudOctomapUpdater::initialize()
   tf_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf_buffer_, root_nh_);
   shape_mask_ = std::make_unique<point_containment_filter::ShapeMask>();
   shape_mask_->setTransformCallback(boost::bind(&PointCloudOctomapUpdater::getShapeTransform, this, _1, _2));
+
+  std::string prefix = "";
+  if (!ns_.empty())
+    prefix = ns_ + "/";
   if (!filtered_cloud_topic_.empty())
-    filtered_cloud_publisher_ = private_nh_.advertise<sensor_msgs::PointCloud2>(filtered_cloud_topic_, 10, false);
+    filtered_cloud_publisher_ =
+        private_nh_.advertise<sensor_msgs::PointCloud2>(prefix + filtered_cloud_topic_, 10, false);
   return true;
 }
 

